### PR TITLE
Fix timeseries chrome overflow

### LINKED ISSUE #42760: The new time granularity option is too intense and overflows
### Describe the bug

Link to the [Slack thread](https://metaboat.slack.com/archives/C02H619CJ8K/p1715773636987059).

> This time granularity option list feels really intense to me and I don’t know why I need to see all of these without disclosing more.

image

### To Reproduce

1. Browse data -> Accounts
2. Add an aggregation on a time column and pick a bucket
3. Open the View ... by ... dropdown underneath the visualization

The dropdown overflows and is quite overwhelming.

### Expected behavior

Make it no overflow and hide some options behind a `More...` button.

### Logs

_No response_

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:125.0) Gecko/20100101 Firefox/125.0",
    "vendor": ""
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.20.1+1",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.20.1",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.20.1+1",
    "os.name": "Mac OS X",
    "os.version": "14.2.1",
    "user.language": "en",
    "user.timezone": "Europe/Amsterdam"
  },
  "metabase-info": {
    "databases": [
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "h2",
    "application-database-details": {
      "database": {
        "name": "H2",
        "version": "2.1.214 (2022-06-13)"
      },
      "jdbc-driver": {
        "name": "H2 JDBC Driver",
        "version": "2.1.214 (2022-06-13)"
      }
    },
    "run-mode": "dev",
    "version": {
      "date": "2024-05-14",
      "src_hash": "0a5efccbcd6508a403492a46be0b5eb713a22d6b",
      "tag": "v0.1.3-SNAPSHOT",
      "hash": "f352fe0"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

### Severity

P3

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by romeovs**
Closes https://github.com/metabase/metabase/issues/42760

### Description

Describe the overall approach and the problem being solved.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Browse data -> Accounts
2. Add an aggregation on a time column and pick a bucket
3. Open the View ... by ... dropdown underneath the visualization

### Demo

Screenshot 2024-05-16 at 14 21 57

### CODE REVIEW

**Review by mazameli — CHANGES_REQUESTED** [review]
Minor, but ideally the `More…` would be `text-brand` in color to distinguish it from the selectable items, and to match what we do in the summarize sidebar and notebook group-by popovers.

File: frontend/src/metabase/querying/components/TimeseriesChrome/TimeseriesBucketPicker/TimeseriesBucketPicker.unit.spec.tsx [hunk 1]
```diff
@@ -87,11 +87,23 @@ describe("TimeseriesBucketPicker", () => {
     expect(getNextBucketName()).toBe("Year");
   });
 
+  it("should allow to show more binning options", async () => {
```
Comment (ranquild) [inline frontend/src/metabase/querying/components/TimeseriesChrome/TimeseriesBucketPicker/TimeseriesBucketPicker.unit.spec.tsx:90]: Please also add tests for the selected item - when it's within the initial expanded list or not

**Review by ranquild — APPROVED** [review]
Looks good, minor comments

**Comment by ranquild** [comment]
@mazameli could you take a look?

**Review by mazameli — APPROVED** [review]

**Pull request merged**